oidc/oidc.go

```diff
@@ -0,0 +1,169 @@
+package oidc
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/gorilla/sessions"
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	"github.com/markbates/goth"
+	"github.com/markbates/goth/gothic"
+	"github.com/markbates/goth/providers/openidConnect"
+)
+
+type OIDC struct {
+	sessionStore *sessions.FilesystemStore
+	provider     *openidConnect.Provider
+
+	name        string
+	baseURL     string
+	signInUrl   string
+	callbackURL string
+}
+
+func SetupOIDC(config Config, baseURL string, e *echo.Echo, skipper middleware.Skipper) (*OIDC, error) {
+	o := &OIDC{
+		name:    "openid-connect",
```

Comment:
only  place where this (const) is used, is in this function, so there's no need for `OIDC.name`.

Same sort of applies to `baseURL`.